lang/src/org/partiql/lang/eval/StrictnessContext.kt

```diff
@@ -0,0 +1,51 @@
+package org.partiql.lang.eval
+
+import org.partiql.lang.ast.MetaContainer
+import org.partiql.lang.errors.ErrorCode
+import org.partiql.lang.errors.PropertyValueMap
+
+
+internal class ErrorDetails(
+    val metas: MetaContainer,
+    val message: String,
+    val errorCode: ErrorCode,
+    val errorContext: PropertyValueMap? = null
+)
+
+internal class StrictnessContext(val strictnessMode: StrictnessMode, val valueFactory: ExprValueFactory) {
+
+    inline fun errorIf(
+        test: Boolean,
+        crossinline detailsConstructor: () -> ErrorDetails,
+        crossinline otherwise: () -> ExprValue
+    ): ExprValue =
+        when {
+            test ->
+                when(strictnessMode) {
+                    StrictnessMode.STANDARD ->
+                        with(detailsConstructor()) {
+                            throw EvaluationException(
+                                message = message,
+                                errorCode = errorCode,
+                                errorContext = errorContext,
+                                cause = null,
+                                internal = false)
+                        }
+                    StrictnessMode.PERMISSIVE ->
+                        valueFactory.missingValue
+                }
```

Comment:
Stylistically, should we just put this as a method on the `enum`, then we don't need an if-bridge clause here, this smells like a factoring that needs some kind of polymorphism either in the `enum` or an algebraic type.  Really the context is just for capturing the evaluation context.  It almost seems like the `enum` is getting in our way in some regards.

Reply:
I'll try again to put these functions as methods on the `ErrorMode` enum but there was a reason I didn't that I can't recall right now...

Reply:
I remember now.  It's because the `ErrorMode` enum is a compiler option, which doesn't seem like the right place for implementation of runtime behavior.  I could split this into two different classes that implement an `ErrorBehavior` interface instead.